pkg/tasks/update_latest_snapshot.go

```diff
@@ -93,28 +93,48 @@ func (t *UpdateLatestSnapshot) Run() error {
 
```

Comment:
would it make sense to extract everything (or almost everything) into a new function on the struct that returned an error?  that way we'd only need to call UpdateLastError once.